tests.js

```diff
@@ -9,9 +9,47 @@
     return;
   }
 
-  function qpair(template, ...values) {
-    return [template, values];
+  // Set up a RegExp subclass so that we can test subclass creation.
+  function SubRegExp(source, flags) {
+    RegExp.call(this, source, flags);
```

Comment:
I believe you'd have to do `var re = new RegExp(source, flags); Object.setPrototypeOf(re, SubRegExp.prototype); return re;` as well as `Object.setPrototypeOf(SubRegExp, RegExp);` ([example](https://github.com/paulmillr/es6-shim/blob/master/test/collections.js#L131-L139))